shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -225,29 +226,46 @@
     composition_order_.clear();
     return did_submit;
   }
+  DetachUnusedLayers();
+  active_composition_order_.clear();
   UIView* flutter_view = flutter_view_.get();
 
-  // This can be more efficient, instead of removing all views and then re-attaching them,
-  // we should only remove the views that has been completly removed from the layer tree, and
-  // reorder the views using UIView's bringSubviewToFront.
-  // TODO(amirh): make this more efficient.
-  // https://github.com/flutter/flutter/issues/23793
-  for (UIView* sub_view in [flutter_view subviews]) {
-    [sub_view removeFromSuperview];
-  }
-
-  active_composition_order_.clear();
   for (size_t i = 0; i < composition_order_.size(); i++) {
     int view_id = composition_order_[i];
-    [flutter_view addSubview:touch_interceptors_[view_id].get()];
-    [flutter_view addSubview:overlays_[view_id]->overlay_view.get()];
+      UIView *intercepter = touch_interceptors_[view_id].get();
+      UIView *overlay = overlays_[view_id]->overlay_view;
+      assert(intercepter.superview  == overlay.superview);
```

Comment:
I think we usually use `NSAssert` in ObjC code? (@cbracken is that correct?)

Reply:
done. Used FML_CHECK after checking with @cbracken offline.